packages/runtime/src/runtime/menu.tsx

```diff
@@ -130,10 +168,80 @@ const CustomLink: React.FC<PropsWithChildren & { item: MenuItem }> = ({
   );
 };
 
-export const SideBarMenu: React.FC<MenuBaseProps> = ({ id, ...props }) => {
-  const [isCollapsed, setIsCollapsed] = useState<boolean>(false);
+export const RenderMenu: React.FC<{
+  type: "SideBar" | "Drawer";
+  menu: MenuBaseProps<SideBarMenuStyles | DrawerMenuStyles>;
+  isCollapsed: boolean;
+  setCollapsed: (collapsed: boolean) => void;
+}> = ({ type, menu, isCollapsed, setCollapsed }) => {
+  return (
+    <div style={{ display: "flex", flexDirection: "row" }}>
+      {type === "SideBar" ? (
+        <SideBarMenu
+          {...menu}
+          isCollapsed={isCollapsed}
+          setCollapsed={setCollapsed}
+        />
+      ) : null}
+      {type === "Drawer" ? (
+        <DrawerMenu {...menu} isOpen={isCollapsed} setOpen={setCollapsed} />
+      ) : null}
+      <div
+        style={{
+          flexGrow: 1,
+          height: "100vh",
+          display: "flex",
+          flexDirection: "column",
+        }}
+      >
+        <Outlet
+          context={
+            type === "Drawer"
+              ? {
+                  openDrawerMenu: () => setCollapsed(true),
+                  closeDrawerMenu: () => setCollapsed(false),
+                }
+              : {
+                  isMenuCollapsed: () => isCollapsed,
+                  setMenuCollapsed: (value: boolean) => setCollapsed(value),
+                }
+          }
+        />
```

Comment:
We can avoid some code dupe here by making the outlet conditional by some property flag.